src/reducers/product-reducer.jsx

```diff
@@ -35,13 +37,36 @@ export const productReducer = (state,action) => {
                 case "Tools":
                     return {...state,category:{...state.category,Tools:!state.category.Tools}}
             }
+
         case "CLEAR_ALL_FILTERS":
             return {
                 ...state,
                 sortBy:"",
                 category:{Plants:false,Planters:false,Fertilizers:false,Tools:false},
                 price:200,
                 rating:0,}
+                
+        case "SET_CART":
+            return {...state,cart:action.payload}
```

Comment:
Here, the same thing is getting returned three times one for the  `SET_CART`, other for  the `ADD_TO_CART`, and for the `REMOVE_FROM_CART`. 

I think, you must be using these actions after a different API call, like to after API call of `add-to-cart` and others.

But as the same state is getting updated after each operation, what you can do na, create only one action may be named `FILL_CART` or `POPULATE_CART` or any other of your choice, and dispatch that action after each API call with the payload of the response.

We usually, create a different action when a different state is getting changed because of that action.

Reply:
Thanks for pointing this out . I have made the changes.